insert-cover-pages-example/src/index.ts

```diff
@@ -0,0 +1,140 @@
+import { MetaContext, createMetaContext } from './Utils/metaContextUtils';
+import { pathCombine } from './Utils/pathUtils';
+import {
+    Command,
+    Commands,
+    createCopyInputPageCommand,
+    createGroupBeginCommmand,
+    createPageSizeCommmand,
+    getCommandFileName,
+} from './Utils/commandUtils';
+
+export function getDescription(): ScriptDescription {
+    return {
+        displayName: 'Insert Cover Pages',
+        description:
+            'Takes a fixed document and adds it to the beginning or end of each document in a record',
+        category: 'Impositioning',
+        input: [
+            parametersDescriptions.InputParamInputFilePath,
+            parametersDescriptions.InputParamMetadataFilePath,
+            parametersDescriptions.InputParamPrefixPagePath,
+            parametersDescriptions.InputParamSufixPagePath,
+            parametersDescriptions.InputParamOutputFilePath,
+            parametersDescriptions.InputParamOutputType,
+            parametersDescriptions.InputParamProductionConfiguration,
+            parametersDescriptions.InputParamWorkingDirectory,
+        ],
+        output: [parametersDescriptions.OutputParamInsertCoverPages],
+    } as const satisfies ScriptDescription;  
+}
+
+export async function execute(context: Context): Promise<Output> {
+    var parameters = Parameters.load(context);
+    Validator.validateParameters(parameters);
+
+    const metadataFile = context.getFile(parameters.metadataFilePath);
+    await Validator.validateMetadataFileExists(metadataFile);
+    const metadataJson = JSON.parse(await metadataFile.read());
+
+    const metaContext = createMetaContext(metadataJson);
+
+    const commandJson = createCommands(
+        metaContext,
+        parameters.inputFilePath,
+        parameters.prefixPagePath,
+        parameters.sufixPagePath
+    );
+    const commandPath = pathCombine(
+        parameters.workingDirectory,
+        getCommandFileName()
+    );
+    await context.write(commandPath, JSON.stringify(commandJson));
+
+    const bundledGenerateStep = await createBundledGenerateOutput(
+        context.environment.impositioningTemplatePath,
+        commandPath,
+        parameters
+    );
+
+    return {
+        insertCoverPages: bundledGenerateStep,
+    };
+}
+
+function createCommands(
+    metaContext: MetaContext,
+    inputFileName: string,
+    inputPrefixPage: string,
+    inputSufixPage: string
+): Commands {
+    const commandJson = {} as Commands;
+    commandJson.pages = [];
+
+    metaContext.groups.forEach((group) => {
+        for (let i = 0; i < group.pageSizes.length; i++) {
+            let newPage: Command[] = [];
+
+            if (i === 0 && inputPrefixPage === undefined || inputPrefixPage === null) {
+                newPage.push(createGroupBeginCommmand());
+                newPage.push(createPageSizeCommmand(group.pageSizes[i]));
+            }
+
+            if (i === 0 && inputPrefixPage != undefined || inputPrefixPage != null) {
```

Comment:
similar to previous bug, replace it by:
if (i===0 && inputPrefixPage != undefined)

Reply:
but actually just move appending of prefix and sufix page outside of this cycle, it will dramaticaly simplify code.